### PR TITLE
Feature: add support for TM4C1294KCDT

### PR DESCRIPTION
**Pull request by desertsagebrush**
## Detailed description
Add support for the TI TM4C1294KCDT processor. This is largely the same as the TM4C1294NCDT, although with less flash (512 KiB versus 1 MiB). With the fixes from earlier, this works on native and hosted/BMDA platforms.

Additionally, I added some extra documentation around some of the preprocessor defines to give a bit more context about what is being masked in the ID registers; I'm glad to drop this if it seems like too much.

## Your checklist for this pull request

* [X] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [X] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [X] It builds for hardware native (`make PROBE_HOST=native`)
* [X] It builds as BMDA (`make PROBE_HOST=hosted`)
* [X] I've tested it to the best of my ability
* [X] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/target/lmi.c
Comment (desertsagebrush) [inline src/target/lmi.c:49]: According to their documentation, Snowflake is the proper family name

Comment (dragonmux) [reply src/target/lmi.c:59]: Sure, not a problem - perhaps make a note of this name mapping in the source as part of the DID0 block comment?

**Review by dragonmux — COMMENTED** [review]
LGTM, we'll merge this once v1.10 is fully out the door. More documentation in-source and more information is always welcome.
Having clear information on how the device ID registers work is a good improvement.

There's one item to answer against your own review note and once handled, we'll approve this ready for the start of the v2.0 cycle as that's not far off now.

**Comment by desertsagebrush** [comment]
Added in documentation of the family name <-> product mapping, as far as I have been able to find. I also found the proper family name for the TM4C123/LM3Fxxx devices (Blizzard), though there might be some other families floating around. I saw some references to 'Tempest' and 'Firestorm' when looking, although I haven't found any datasheets referencing those family names.

**Review by dragonmux — APPROVED** [review]
LGTM! At least now, if/when those other families come to light to be added, there's some place to add them and flesh out the naming mapping.

**Pull request merged**
